web-extension/common/backend/popup-client.js

```diff
@@ -0,0 +1,56 @@
+import JSONRPCClient from '../lib/json-rpc-client.js'
+import assert from 'nanoassert'
+
+export class PopupClient {
+  constructor() {
+    this.ports = new Set()
+    this.persistentQueue = []
+
+    this.client = new JSONRPCClient({
+      idPrefix: 'background-',
+      send: (msg) => {
+        this.persistentQueue.push(msg)
+        this.ports.forEach((p) => p.postMessage(msg))
+      }
+    })
+  }
+
+  async reviewConnection(params) {
+    return this.client.request('popup.review_connection', params)
+  }
+
+  async reviewTransaction(params) {
+    return this.client.request('popup.review_transaction', params)
+  }
+
+  async connect(port) {
```

Comment:
So I think we should be unit testing this too unless I am missing something. Checking onMessage and onDisconnect functionality works as expected. Saves letting these errors bubble upwards and having to be troubleshooted further up the stack